packages/cli/src/commands/hydrogen/env/push__unstable.ts

```diff
@@ -0,0 +1,259 @@
+      validatedEnvironment = {id, name, branch, type};
+    } else {
+      // Prompt the user for a selection if there are multiple matches
+      const selection = await renderSelectPrompt({
+        message: `There were multiple environments found with the name ${environmentName}:`,
+        choices: [
+          ...matchedEnvironments.map(({id, name, branch, type, url}) => ({
+            label: `${name} (${branch}) ${type} ${url}`,
+            value: id,
+          })),
+        ],
+      });
+      const {id, name, branch, type} =
+        matchedEnvironments.find(({id}) => id === selection) ?? {};
+      validatedEnvironment = {id, name, branch, type};
+    }
+  } else {
+    // Environment flag not passed
+    const choices = [
+      ...environments.map(({id, name, branch}) => ({
+        label: branch ? `${name} (${branch})` : name,
+        value: id,
+      })),
+    ];
+
+    const pushToBranchSelection = await renderSelectPrompt({
+      message: 'Select a set of environment variables to overwrite:',
+      choices,
+    });
+
+    const {id, name, branch, type} =
+      environments.find(({id}) => id === pushToBranchSelection) ?? {};
+    validatedEnvironment = {id, name, branch, type};
+  }
+
+  // Fetch remote variables
+  const {environmentVariables = []} =
+    (await getStorefrontEnvVariables(
+      session,
+      config.storefront.id,
+      validatedEnvironment.branch ?? undefined,
+    )) ?? {};
+
+  // Normalize variables
+  const remoteVars = environmentVariables.filter(
+    ({isSecret, readOnly}) => !isSecret && !readOnly,
+  );
+  const comparableRemoteVars =
+    remoteVars
+      .sort((a, b) => a.key.localeCompare(b.key))
+      .map(({key, value}) => `${key}=${value}`)
+      .join('\n') + '\n';
+
+  const compareableLocalVars =
+    Object.keys(localVariables)
+      .sort((a, b) => a.localeCompare(b))
+      .reduce((acc, key) => {
+        const {isSecret, readOnly} =
+          environmentVariables.find((variable) => variable.key === key) ?? {};
+        if (isSecret || readOnly) return acc;
+        return [...acc, `${key}=${localVariables[key]}`];
+      }, [] as string[])
+      .join('\n') + '\n';
+
+  // Confirm changes and show a generate diff of changes
+  if (!validatedEnvironment.name)
+    throw new AbortError('Missing environment name');
+
+  // Generate a list of remote read only or secrets
+  const remoteReadOnlyOrSecrets = environmentVariables.reduce(
+    (acc, {isSecret, readOnly, key}) => {
+      if (!isSecret && !readOnly) return acc;
+      const localVar = localVariables[key];
+      const remoteVar = environmentVariables.find(
+        (variable) => variable.key === key,
+      );
+      if (localVar === remoteVar?.value) return acc;
+      return [...acc, key];
+    },
+    [] as string[],
+  );
+
+  if (remoteReadOnlyOrSecrets.length) {
+    outputWarn(
+      `\n\nVariables that are read only or contain secret values cannot be pushed from the CLI: ${remoteReadOnlyOrSecrets.join(
```

Comment:
You can remove one of the newlines in the beginning. There is a LOT of spacing in the outputWarn.

<img width="915" alt="image" src="https://github.com/Shopify/hydrogen/assets/2907059/6ed578d9-78a2-4894-a1b7-fc12fcf3316f">

Reply:
I think I can remove both! Will do.